scripts/seedUser.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const csvParser = require('csv-parser');
+const { Client } = require('pg');
+const moment = require('moment');
+
+
+console.log('Getting postgres URL');
+const env = process.env.NODE_ENV || 'development';
+// eslint-disable-next-line import/no-dynamic-require
+const config = require(`${__dirname}/../config/config.js`)[env];
+
+if (!config.url) {
+  console.error('Postgres url not found. Please ensure you set one in your env.');
+  process.exit(1);
+}
+
+const filePath = process.argv[2];
+
+const client = new Client({
+  connectionString: config.url,
+});
+
+function seedData(data) {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const [index, item] of data.entries()) {
+    const id = parseInt(item.membershipNo, 10);
+    const date = moment().format('YYYY-MM-DD HH:mm');
+    client.query(
+      `
```

Comment:
- What advantage does parameterised query has over template strings?
- Is it possible to completely eliminate using a loop? Would be great if you can provide a link or resource I could check out.
- Why do we need to pass in the password as an argument? The only reason why I included the password is because it's a required field. The user will still need to register with a new password before they can use the platform.
- This is not a regular seed file. Seeding can only be performed by some set of people that have access to the original csv file. The csv file included is just a sample for people to test with on their local.
- I didn't want to go through the stress of hashing the password considering it's not a real password and won't be giving out to the users. It's just there to fulfil all righteousness 😄 


```
$1 = comma separated column names
$2 = comma separated values e.g (1,tony,stark,edward,hashedpassword,date,date),(2,bruce,banner,robert,hashedpassword,date,date)
```

I get this. This makes sense if it has a significance advantage over the crud way I used before. But this won't eliminate the use of a forloop. We'll still need it to prepare the values.
 
```
also chances are you do not need to use a csv parser, you can maybe use readline, if you want to run a query - https://nodejs.org/api/readline.html
```

-  can you point me to an example where that is used? The documentation shows that it's used to get feedback from users using cli.

Reply:
- parameterised query automatically escapes the input, and is best practice when using ORMs or library to run SQL statements. (just in case the input has characters that are likely to cause an expected query to run)
- if we do this `'INSERT INTO "Users" ($1) VALUES ($2) ON CONFLICT DO NOTHING;'` we eliminate the loop, we just need to have a formatted string (this will happen in readFile) that fits into the SQL statement

the last code snippet here
https://nodejs.org/api/readline.html#readline_example_read_file_stream_line_by_line